cappy.py

```diff
@@ -15,41 +15,53 @@
 
 
 class Promise(object):
+    """TODO: Docstring."""
+
     def __init__(self):
+        """TODO: Docstring."""
         self.callbacks = []
 
     def add_callback(self, callback):
+        """TODO: Docstring."""
         self.callbacks.append(callback)
 
     def fufill(self, arg):
+        """TODO: Docstring."""
         result = arg
         for callback in self.callbacks:
             result = callback(result)
 
 
 class Obligation(object):
+    """TODO: Docstring."""
+
     def __init__(self, promise):
+        """TODO: Docstring."""
         self.promise = promise
 
     def call(self, arg):
+        """TODO: Docstring."""
         self.promise.fufill(arg)
 
 
 def make_promise():
+    """TODO: Docstring."""
     p = Promise()
     o = Obligation(p)
     return p, o
 
 
 class JsonDataStream(object):
-    """Consumes bytes and produces JSON objects"""
+    """Consumes bytes and produces JSON objects."""
 
     def __init__(self):
+        """TODO: Docstring."""
         self.depth = 0
         self.buf = ''
 
     def receive(self, data):
-        """Process incoming bytes, returning a json object if 
+        # XXX: This docstring sounds weird, if what?
+        """Process incoming bytes, returning a json object if.
```

Comment:
Let's just fix the docstring. It should say that the method returns a list of dicts, each one representing a json object from the wire. If no complete json objects are contained in the bytes from the wire, the list is empty.